portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,41 +38,38 @@
 
   private final Gson gson = new Gson();
 
-  /**
-   * Gets the comments from datastore
-   * @param request - client asks information from servlet
-   * @param response - servlet writes response
-   * @return send the response back to client
-   */
   @Override
   public void doGet(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
 
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
+    // Limit on the number of comments to get from datastore.
+    String userLimit = request.getParameter("limit");
+    FetchOptions limit;
+
+    if(userLimit == null || userLimit.isEmpty()) {
+       limit = FetchOptions.Builder.withLimit(results.countEntities(FetchOptions.Builder.withDefaults()));
+    } else {
+       limit = FetchOptions.Builder.withLimit(Integer.parseInt(userLimit));
```

Comment:
parseInt can throw an error if the user has put something dumb in as the limit (always assume your user will do something silly). This should probably be wrapped in a try catch.

Reply:
Done :)